app/api/groups/create_group.go

```diff
@@ -0,0 +1,148 @@
+	"github.com/France-ioi/AlgoreaBackend/app/service"
+)
+
+// swagger:model createGroupRequest
+type createGroupRequest struct {
+	// required: true
+	// minLength: 1
+	Name *string `json:"name" validate:"required,min=1"`
+	// required: true
+	// enum: Class,Team,Club,Friends,Other
+	Type *string `json:"type" validate:"required,oneof=Class Team Club Friends Other"`
+	// only if `type` = "Team"
+	// required: true
+	// type: string
+	// format: int64
+	// minimum: 1
+	ItemID *int64 `json:"item_id,string" validate:"min=1"`
+}
+
+// swagger:operation POST /groups groups groupCreate
+// ---
+// summary: Create a group
+// description: >
+//
+//   Creates a group with the input `name`, `type`, `sDateCreated` = now(), and default values in other columns.
+//   If `item_id` is given:
+//
+//     * If `type` != "Team", returns the "badRequest" response
+//
+//     * Otherwise, checks that the authenticated user
+//
+//       * has grayed, partial or full access to the item (otherwise returns the "forbidden" response)
+//
+//       * sets this `item_id` as `idTeamItem` of the new group.
+//
+//   Also, the service sets the authenticated user as an owner of the group (with `sRole` = "owner").
+//   After everything, it propagates group ancestors.
+//
+//
+//   The user should not be temporary, otherwise the "forbidden" response is returned.
+// consumes:
+// - application/json
+// parameters:
+// - in: body
+//   name: data
+//   required: true
+//   description: The group to create
+//   schema:
+//     "$ref": "#/definitions/createGroupRequest"
+// responses:
+//   "201":
+//     description: "Created. Success response with the created group's ID"
+//     schema:
+//       type: object
+//       required: [success, message, data]
+//       properties:
+//         success:
+//           description: "true"
+//           type: boolean
+//           enum: [true]
+//         message:
+//           description: created
+//           type: string
+//           enum: [created]
+//         data:
+//           type: object
+//           required: [id]
+//           properties:
+//             id:
+//               type: string
+//               format: int64
+//   "400":
+//     "$ref": "#/responses/badRequestResponse"
+//   "401":
+//     "$ref": "#/responses/unauthorizedResponse"
+//   "403":
+//     "$ref": "#/responses/forbiddenResponse"
+//   "500":
+//     "$ref": "#/responses/internalErrorResponse"
+func (srv *Service) createGroup(w http.ResponseWriter, r *http.Request) service.APIError {
+	var err error
+	user := srv.GetUser(r)
+
+	input := createGroupRequest{}
+	formData := formdata.NewFormData(&input)
+	err = formData.ParseJSONRequestData(r)
+	if err != nil {
+		return service.ErrInvalidRequest(err)
+	}
+
+	if input.ItemID != nil && *input.Type != "Team" {
+		return service.ErrInvalidRequest(errors.New("only teams can be created with item_id set"))
+	}
+
+	if user.SelfGroupID == nil || user.OwnedGroupID == nil {
```

Comment:
The idea was to prevent users having idGroupSelf/idGroupOwned from creating groups. Actually it is not about temp users (although temp user have idGroupOwned = NULL & idGroupSelf != NULL).
Other services are all right because `... = NULL` never gives true in MySQL. Also, we check SelfGroupID & OwnedGroupID where needed before using them.

Reply:
But then do you actually check that it is not a temp user? or is it just a side effect of that?
Anyway, you should comment this kind of stuffs, this "if" condition is a bit too abstract.

Reply:
added more comments

Reply:
But one of my previous comments still stands: 

> As there is `tempUser` for that, use this attribute in priority (as one may choose to give them an ownerdgroupid without knowing he has to fix this service).

So I think you should check tempUser attribute explicitly as well (as it is free as you already get some other user info)

Reply:
fixed